cdap-docs/application-templates/source/etl/templates/sources/real-time/kafka.rst

```diff
@@ -6,6 +6,36 @@
 Sources: Real-time: Kafka 
 ===============================
 
-.. rubric:: Description: Kafka Realtime Source
+.. rubric:: Description
 
-Emits a record with two fields - 'key' (nullable string) and 'message' (bytes).
+Kafka Realtime Source. Emits a record with two fields - 'key' (nullable string) and 'message' (bytes).
+
+.. rubric:: Use Case
+
+TODO: Fill me out
+
+.. rubric:: Properties
+
+**kafka.partitions:** Number of partitions.
+
+**kafka.topic:** Topic of the messages.
+
+**kafka.zookeeper:** The connect string location of Zookeeper.
+Either this or the list of brokers is required.
+
+**kafka.brokers:** Comma separated list of Kafka brokers. Either this or the Zookeeper connect info is required.
+
+**kafka.default.offset:** The default offset for the partition. Default value is kafka.api.OffsetRequest.EarliestTime.
+
+**kafka.schema:** Optional schema for the body of Kafka events.
+Schema is used in conjunction with format to parse Kafka payloads. Some formats like the 'avro' format require schema,
```

Comment:
The schema is used in conjunction with the format to parse Kafka payloads. Some formats (such as the 'avro' format) require schema while others do not. The schema given is for the body of the Kafka event.